code/datums/components/deployable_item.dm

```diff
@@ -89,10 +89,15 @@
 			if(!placement_loc.allow_construction || area.area_flags & NO_CONSTRUCTION) // long ass series of checks to prevent things like deployable shields on alamo
 				user.balloon_alert(user, "Can't deploy here")
 				return
-
 		if(LinkBlocked(get_turf(user), location))
 			location.balloon_alert(user, "No room to deploy")
 			return
+		if(istype(item_to_deploy, /obj/item/weapon/gun/sentry))
+			var/sentry_range = orange(TURRET_DEPLOY_RANGE_RESTRICTION, location)
+			if(locate(/obj/machinery/deployable/mounted/sentry) in sentry_range)
+				if(!locate(/obj/structure/dropship_equipment/shuttle/sentry_holder) in sentry_range)
+					location.balloon_alert(user, "Another sentry is too close")
+					return
```

Comment:
This is very odd, I'd rework to

```suggestion
			for(var/obj/machinery/deployable/mounted/sentry/found_sentry in nearby_objects)
				if(isturf(found_sentry.loc)))
					var/obj/item/weapon/gun/sentry/internal_sentry = found_sentry.get_internal_item()
					if(internal_sentry.item_flags & IS_DEPLOYED)
						location.balloon_alert(user, "Another sentry is too close!")
```

Be sure to test this and modify if I made a mistake. But this should cover what you want to do. Goes for the second instance of this loop as well of course.